packages/scripts/src/tasks/providers/android.ts

```diff
@@ -70,86 +70,28 @@ const configure = async ({ appId, appName, versionName }: IAndroidBuildOptions)
   }
 };
 
-const set_splash_image = async (splashAssetPath: string) => {
-  if (!fs.existsSync(splashAssetPath)) {
-    return Logger.error({
-      msg1: "Splash source image not found",
-      msg2: `A source .png file is required to generate a splash screen. No asset was found at the path supplied in the deployment config: ${splashAssetPath}.`,
-    });
-  }
-
-  const cordovaOptions: Options = {
-    directory: ROOT_DIR,
-    resourcesDirectory: join(ROOT_DIR, "resources"),
-    logstream: process.stdout,
-    platforms: {
-      android: {
-        splash: {
-          sources: [splashAssetPath],
-        },
-      },
-    },
-    skipConfig: true,
-    copy: true,
-    projectConfig: {
-      android: {
-        directory: join(ROOT_DIR, "android"),
-      },
-    },
-  };
+const set_icons_and_splash_images = async (options: { assetPath: string }) => {
+  const { assetPath } = options;
 
-  return await run(cordovaOptions);
-};
-
-const set_launcher_icon = async (options: {
-  iconAssetPath: string;
-  iconAssetForegroundPath?: string;
-  iconAssetBackgroundPath?: string;
-}) => {
-  const { iconAssetPath, iconAssetForegroundPath, iconAssetBackgroundPath } = options;
-
-  const iconSources = [];
-  if (fs.existsSync(iconAssetPath)) {
-    iconSources.push(iconAssetPath);
+  const iconAndSplashSources = [];
+  if (fs.existsSync(assetPath)) {
+    iconAndSplashSources.push(assetPath);
   } else {
     return Logger.error({
-      msg1: "Icon source asset not found",
-      msg2: `A source .png file is required to be used as a fall back for when the device's android version does not support adaptive icons. No asset was found at the path supplied in the deployment config: ${iconAssetPath}.`,
+      msg1: "Launcher icon and splash images source assets not found",
+      msg2: `No folder was found at the path supplied in the deployment config: ${assetPath}.`,
     });
   }
 
-  const includeAdaptiveIcons =
-    fs.existsSync(iconAssetForegroundPath) && fs.existsSync(iconAssetBackgroundPath);
-
-  const cordovaOptions: Options = {
-    directory: ROOT_DIR,
-    resourcesDirectory: join(ROOT_DIR, "resources"),
-    logstream: process.stdout,
-    platforms: {
-      android: includeAdaptiveIcons
-        ? {
-            "adaptive-icon": {
-              icon: { sources: iconSources },
-              foreground: { sources: [iconAssetForegroundPath] },
-              background: { sources: [iconAssetBackgroundPath] },
-            },
-          }
-        : { icon: { sources: iconSources } },
-    },
-    skipConfig: true,
-    copy: true,
-    projectConfig: {
-      android: {
-        directory: join(ROOT_DIR, "android"),
-      },
-    },
-  };
+  // Generate android assets
+  const assetDirPath = assetPath.includes(".png") ? path.dirname(assetPath) : assetPath;
```

Comment:
Do we know whether the file has to be a `.png`? Either way, I think it would be better to check whether the file is to a path in a more general way. Have a look at [these docs](https://www.geeksforgeeks.org/how-to-check-the-given-path-is-file-or-directory-in-node-js/) (we're already importing `fs` so can use its methods)